### PR TITLE
ROU-4045: Add Size feature to ProgressCircle

### PR DESCRIPTION
**Pull request by bmarcelino-fe**
This PR is for adding the Size parameter to ProgressCircle. Check the JIRA issue for more information.

### Checklist
-   [X] tested locally
-   [X] documented the code
-   [X] clean all warnings and errors of eslint
-   [ ] requires changes in OutSystems (if so, provide a module with changes)
-   [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

**Review by JoaoFerreira-FrontEnd — APPROVED** [review]

**Comment by BenOsodrac** [comment]
@bmarcelino-fe dist complied files shouldn't be commited. 

@joselrio Again, shouldn't these be added to the git ignore?

File: src/scripts/OSFramework/OSUI/Pattern/Progress/Circle/ProgressCircle.ts [hunk 1]
```diff
@@ -44,27 +44,40 @@ namespace OSFramework.OSUI.Patterns.Progress.Circle {
...
-			// Get the pattern parent size
-			const _elementSize =
-				this.selfElement.parentElement.clientHeight < this.selfElement.parentElement.clientWidth
-					? this.selfElement.parentElement.clientHeight
-					: this.selfElement.parentElement.clientWidth;
-
-			// Check the maxValue that the circle must have
-			if (this.selfElement.clientHeight < this.selfElement.parentElement.clientWidth) {
-				this._circletSize = this.selfElement.parentElement.clientWidth;
+			// Check wich size will be appliced on ProgressCircle
```
Comment (BenOsodrac) [inline src/scripts/OSFramework/OSUI/Pattern/Progress/Circle/ProgressCircle.ts:47]: typo on applied

File: src/scripts/OSFramework/OSUI/Pattern/Progress/Circle/ProgressCircle.ts [hunk 2]
```diff
@@ -44,27 +44,40 @@ namespace OSFramework.OSUI.Patterns.Progress.Circle {
...
+			if (this.configs.ProgressCircleSize !== '') {
```
Comment (BenOsodrac) [inline src/scripts/OSFramework/OSUI/Pattern/Progress/Circle/ProgressCircle.ts:48]: This validation will create a strange behaviour with the following steps on the QA test screen:

1. Set width to something, like 100vw and apply. Works correctly ✅ 
2. Set width back to empty (which would be the default behavior of parent size). The size doesn't update, as the css variable isn't updated, due to this line.

Maybe the validation should be something like this:
`this.configs.ProgressCircleSize !== '' ||  this.configs.ProgressCircleSize === '' && Helper.Dom.Styles.GetStyleAttribute(this.selfElement,Enum.InlineStyleProp.ProgressCircleSize) !== '100%';`

**Review by BenOsodrac — CHANGES_REQUESTED** [review]

File: src/scripts/OSFramework/OSUI/Pattern/Progress/Circle/ProgressCircle.ts (hunk 2 above)
Comment (bmarcelino-fe) [reply src/scripts/OSFramework/OSUI/Pattern/Progress/Circle/ProgressCircle.ts:48]: Fixed

File: src/scripts/OSFramework/OSUI/Pattern/Progress/Circle/ProgressCircle.ts (hunk 1 above)
Comment (bmarcelino-fe) [reply src/scripts/OSFramework/OSUI/Pattern/Progress/Circle/ProgressCircle.ts:47]: fixed

**Review by BenOsodrac — APPROVED** [review]

**Review by JoaoFerreira-FrontEnd — APPROVED** [review]

**Pull request merged**